go/controller/post_token_test.go

```diff
@@ -0,0 +1,238 @@
+						"x-hasura-user-id":           "db477732-48fa-4289-b694-2886a646b6eb",
+						"x-hasura-user-is-anonymous": "false",
+					},
+					"iat": float64(time.Now().Unix()),
+					"iss": "hasura-auth",
+					"sub": "db477732-48fa-4289-b694-2886a646b6eb",
+				},
+				Signature: []byte{},
+				Valid:     true,
+			},
+			customClaimer: nil,
+			emailer:       nil,
+			hibp:          nil,
+			jwtTokenFn:    nil,
+		},
+		{
+			name: "invalid refresh token",
+			config: func() *controller.Config {
+				cfg := getConfig()
+				cfg.AllowedEmails = []string{"asd@asd.com"}
+				return cfg
+			},
+			db: func(ctrl *gomock.Controller) controller.DBClient {
+				mock := mock.NewMockDBClient(ctrl)
+
+				mock.EXPECT().GetUserByRefreshTokenHash(
+					gomock.Any(),
+					sql.GetUserByRefreshTokenHashParams{
+						RefreshTokenHash: sql.Text(hashedToken),
+						Type:             "regular",
+					},
+				).Return(sql.AuthUser{}, pgx.ErrNoRows) //nolint:exhaustruct
+
+				return mock
+			},
+			customClaimer: nil,
+			request: api.PostTokenRequestObject{
+				Body: &api.RefreshTokenRequest{
+					RefreshToken: token.String(),
+				},
+			},
+			expectedResponse: controller.ErrorResponse{
+				Error:   "invalid-refresh-token",
+				Message: "Invalid or expired refresh token",
+				Status:  401,
+			},
+			expectedJWT: nil,
+			emailer:     nil,
+			hibp:        nil,
+			jwtTokenFn:  nil,
+		},
+		{
+			name:   "user disabled",
+			config: getConfig,
+			db: func(ctrl *gomock.Controller) controller.DBClient {
+				mock := mock.NewMockDBClient(ctrl)
+
+				user := getSigninUser(userID)
+				user.Disabled = true
+
+				mock.EXPECT().GetUserByRefreshTokenHash(
+					gomock.Any(),
+					sql.GetUserByRefreshTokenHashParams{
+						RefreshTokenHash: sql.Text(hashedToken),
+						Type:             "regular",
+					},
+				).Return(user, nil)
+
+				return mock
+			},
+			customClaimer: nil,
+			request: api.PostTokenRequestObject{
+				Body: &api.RefreshTokenRequest{
+					RefreshToken: token.String(),
+				},
+			},
+			expectedResponse: controller.ErrorResponse{
+				Error:   "disabled-user",
+				Message: "User is disabled",
+				Status:  401,
+			},
+			expectedJWT: nil,
+			emailer:     nil,
+			hibp:        nil,
+			jwtTokenFn:  nil,
+		},
+		{
+			name:   "user anonymous",
+			config: getConfig,
+			db: func(ctrl *gomock.Controller) controller.DBClient {
+				mock := mock.NewMockDBClient(ctrl)
+
+				user := getSigninUser(userID)
+				user.IsAnonymous = true
+
+				mock.EXPECT().GetUserByRefreshTokenHash(
+					gomock.Any(),
+					sql.GetUserByRefreshTokenHashParams{
+						RefreshTokenHash: sql.Text(hashedToken),
+						Type:             "regular",
+					},
+				).Return(user, nil)
+
+				return mock
+			},
+			customClaimer: nil,
+			request: api.PostTokenRequestObject{
+				Body: &api.RefreshTokenRequest{
+					RefreshToken: token.String(),
+				},
+			},
+			expectedResponse: controller.ErrorResponse{
+				Error:   "forbidden-anonymous",
```

Comment:
do we want this behaviour? I haven't checked what we used to do in the nodejs version but I believe an anonymous user should be able to refresh the token